File: C++/is_prime.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isPrime(int x)
{
    if (x == 2)
        return(true);
    else if (x % 2 == 0 or x < 2)
        return(false);
    else
    {
        for (int i = 3; i <= (int)sqrt((double)x); i++)
            if (x % i == 0)
                return(false);
        return(true);
    }
}

int main()
{
    int x;
    
    cin >> x;
    
    if (isPrime(x))
        cout << "Yes" << endl;
    else
        cout << "No" << endl;
    return 0;
}

File: Lua/is_prime.lua
function isPrime(x)
  if x == 2 then
    return true
  elseif x % 2 == 0 or x < 2 then
    return false
  else
    for i=3,math.sqrt(x) do
      if x % i == 0 then
        return false
      end
    end
    return true
  end
end

local x = tonumber(io.read())

if isPrime(x) then
  print("Yes")
else
  print("No")
end
